docs/collections/documents/tutorials/using_rest_api.md

```diff
@@ -922,189 +1090,215 @@ print("Delete query deleted: ", resp.text)
+    )
+    .catch((error) => console.log(error));
+}
 
-      const removeResults = await execute();
+async function updateSavedQueryWorker (queryName, newQuery, queryParams) {
+  await connection
+    .req(`/_fabric/_system/_api/restql/${queryName}`, {
+      body: {
+        query: {
+          value: newQuery,
+          parameter: queryParams
+        }
+      },
+      method: "PUT"
+    })
+    .then((updatedQuery) =>
+      console.log(`${queryName} was modified successfully`, updatedQuery)
+    )
+    .catch((error) => console.log(error));
+}
 
-      console.log("DATA AFTER DELETE", removeResults);
+async function deleteQueryWorker (queryName) {
+  await connection
+    .req(`/_fabric/_system/_api/restql/${queryName}`, {
+      method: "DELETE"
+    })
+    .then((query) => console.log("Query deleted successfully", query))
+    .catch((error) => console.log(error));
+}
 
-      /* ----------------------------- Delete RestQL query by name ----------------------------- */
+async function createCollection (collection) {
+  await connection
+    .req("/_fabric/_system/_api/collection", {
+      body: { name: collection },
+      method: "POST"
+    })
+    .then((collection) =>
+      console.log("2. Collection saved successfully", collection)
+    )
+    .catch((error) => console.log(error));
+}
 
-      const deleteQuery = await connection.req(
-        `/_fabric/_system/_api/restql/${QUERY_NAME}`,
-        {
-          method: "DELETE",
-        }
-      );
+async function deleteCollection (collection) {
+  await connection
+    .req(`/_fabric/_system/_api/collection/${collection}`, {
+      method: "DELETE"
+    })
+    .then((collection) =>
+      console.log("Collection deleted successfully", collection)
+    )
+    .catch((error) => console.log(error));
+}
 
-      console.log("QUERY DELETED  SUCCESSFULLY", deleteQuery);
-    } catch (e) {
-      console.error(e);
-    }
-  };
+const run = async function () {
+  try {
+    /* -------------------- Log in (nemo@nautilus.com/xxxxxx) -------------------- */
+
+    await connection
+      .login(email, password)
+      .then(() => console.log("\n1. User authentication done!"))
+      .catch((error) => console.log(error));
+
+    /* ------------------------ Create Collection ----------------------- */
+    await createCollection(collectionName);
+
+    console.log("3. Saving query workers");
+
+    /* ------------------------ Save query worker  ----------------------- */
+    await saveQueryWorker(readQueryName, readQuery, queryParams);
+
+    /* -------------------  Insert query worker ------------------- */
+    await saveQueryWorker(insertQueryName, insertQuery, queryParams);
+    /* -------------------  Update query worker ------------------- */
+    await saveQueryWorker(updateQueryName, updateQuery, queryParams);
+
+    /* -------------------  Delete query worker ------------------- */
+    await saveQueryWorker(deleteQueryName, deleteQuery, queryParams);
+
```

Comment:
@edgargarciamacrometa you missed adding a small sleep(wait time) here between saving a query worker and executing it(This is required as there is a tiny bit of lag between saving query worker and having it ready for execution). You can use something like this `await new Promise(r => setTimeout(r, 2000));` It will add a 2sec wait time.

Reply:
Also I see two pull requests for the same thing PR-263 (this) and PR-262?